### PR TITLE
support FIONBIO/MSG_DONTWAIT

### PR DESCRIPTION
**Pull request by anchao**
time/gmtimer: remove unnecessary definition
stdio/getline: correct the character slash
stdio/getline: enable getline in either CRLF
Kconfig: add PATH_MAX config
time: nest sys/time.h to avoid undefined types
type/select: add select fd_set to types include
net/socket: add MSG_DONTWAIT support
net/ioctl: add FIONBIO support
sys/file: dummy header for POSIX compliance
fcntl: add O_CLOEXEC flag

### CODE REVIEW

File: include/sys/file.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/********************************************************************************
+ * include/sys/file.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *   http://www.apache.org/licenses/LICENSE-2.0
```
Comment (patacongo) [inline include/sys/file.h]: Need a blank line

File: include/sys/types.h [hunk 2]
```diff
@@ -301,6 +301,9 @@ typedef CODE int (*main_t)(int argc, FAR char *argv[]);
 
 #endif /* __ASSEMBLY__ */
 
+/* Defines `fd_set' and the FD_* macros for `select'. */
+#include <sys/select.h>
```
Comment (patacongo) [inline include/sys/types.h]: All comments must be followed by a blank line

File: include/time.h [hunk 3]
```diff
@@ -240,6 +240,8 @@ int nanosleep(FAR const struct timespec *rqtp, FAR struct timespec *rmtp);
 void tzset(void);
 #endif
 
+#include <sys/time.h>
+
```
Comment (patacongo) [inline include/time.h]: This seems suspicious.  These are separate files with separate requirements and definitions.  Merging this does not seem to be in the spirit of the definitions of OpenGroup.org

Comment (patacongo) [inline include/time.h]: This seems suspicious.  These are separate files with separate requirements and definitions.  Merging this does not seem to be in the spirit of the definitions of OpenGroup.org

Comment (patacongo) [inline include/time.h]: This seems suspicious.  These are separate files with separate requirements and definitions.  Merging this does not seem to be in the spirit of the definitions of OpenGroup.org

File: include/sys/file.h (hunk 1 above)
Comment (davids5) [reply include/sys/file.h]:
```suggestion
 *   http://www.apache.org/licenses/LICENSE-2.0
 
```

File: sched/Kconfig [hunk 4]
```diff
@@ -1116,6 +1116,12 @@ config NAME_MAX
 	---help---
 	The maximum size of a file name.
 
+config PATH_MAX
+	int "Maximum size of path name"
+	default 256
+	---help---
+	The maximum size of path name.
```
Comment (patacongo) [inline sched/Kconfig]: Missing TAB indentation

File: libs/libc/stdio/lib_getdelim.c [hunk 5]
```diff
@@ -53,7 +53,7 @@
  */
 
 #undef HAVE_GETLINE
-#if defined(CONFIG_EOL_IS_CR)
+#if defined(CONFIG_EOL_IS_CR) || defined(CONFIG_EOL_IS_EITHER_CRLF)
```
Comment (patacongo) [inline libs/libc/stdio/lib_getdelim.c]: Does this work?  Can getline() handle two character EOL sequences?

File: libs/libc/time/lib_gmtimer.c [hunk 6]
```diff
@@ -51,10 +51,6 @@
  * Pre-processor Definitions
  ****************************************************************************/
```
Comment (patacongo) [inline libs/libc/time/lib_gmtimer.c]: If the "Pre-processor Definitions" section is now empty, then the preceding block comment should also be removed.

File: libs/libc/stdio/lib_getdelim.c (hunk 5 above)
Comment (anchao) [reply libs/libc/stdio/lib_getdelim.c]: Yes, the processing logic just follows other similar code in apps cle, see:
apps/system/cle/cle.c:
`1108         /* Newline terminates editing.  But what is a newline? */
1109 
1110 #if defined(CONFIG_EOL_IS_CR) || defined(CONFIG_EOL_IS_EITHER_CRLF)
1111         case '\r': /* CR terminates line */
1112 `

i think in either case we can handler by CR in geline by default

**Comment by patacongo** [comment]
Sorry, I didn't mean to close this.  I often hit the wrong button.  (Then it had to stay closed for a long time because of Github problems.. It would not let me comment or re-open the PR)

File: libs/libc/stdio/lib_getdelim.c (hunk 5 above)
Comment (patacongo) [reply libs/libc/stdio/lib_getdelim.c]: I don't think so.  The /n will not be consumed but will come through as data, won't it?

Comment (anchao) [reply libs/libc/stdio/lib_getdelim.c]: Ok, I will consider a better solution,  new change will upload later, please close this PR, thanks for review!

**Comment by patacongo** [comment]
Replaced with other PRs.

**Pull request closed without merging**